app/main/posts/modify/post-data-editor.directive.js

```diff
@@ -101,6 +101,11 @@ function PostDataEditorController(
             $scope.cancel();
         }
     });
+    window.onbeforeunload = function (e) {
+        if ($scope.postForm && $scope.postForm.$dirty) {
+            return 'Are you sure you want to leave this page?  You will lose any unsaved data.';
```

Comment:
@Angamanga create a leavePost() function to handle this, ideally we need to call through to that
1. For consistency
2. For translation of the message.

https://github.com/ushahidi/platform-client/blob/37395583ddec3e845b855342edbc4a2f62f7d4f7/app/main/posts/modify/post-data-editor.directive.js#L97